packages/language-json/package.json

```
@@ -11,5 +11,25 @@
   "license": "MIT",
   "dependencies": {
     "tree-sitter-json": "0.20.0"
+  },
+  "configSchema": {
+    "allowCommentsInJsonFiles": {
+      "type": "boolean",
+      "title": "Allow Comments in .json files",
+      "description": "When checked, will allow comment syntax in all files with the `.json` extension even though it’s not valid JSON. When unchecked, comments in `.json` files will be marked as invalid, and only files with the `.jsonc` extension will be allowed to show comments. (It’s recommended to reload your window or restart Pulsar after changing this setting.)",
```

I wonder if a less-technical-user friendlier explanation is possible here?

This is primarily for things like highlighting, code folding, etc, right?

So I might rephrase some of it?

```suggestion
      "description": "When checked, will allow comments in `.json` files, even though it’s not valid according to the official JSON specification. When unchecked, comments in `.json` files will be marked as invalid, and only files with the `.jsonc` extension will mark and highlight comments as normal/allowed. (It’s recommended to reload your window or restart Pulsar after changing this setting.)",
```

If this is off-base, then feel free to not change the wording, suggestion was to give an example of how I thought it might be made more approachable, but I am not attached to the specific phrasing in the code suggestion.

---

I think that's fine, and I'll tweak it only slightly from what you've suggested:

> When checked, will allow comments in `.json` files even though they’re not valid according to the official JSON specification. When unchecked, comments in `.json` files will be marked as invalid, and only files with the `.jsonc` extension will mark comments as valid and allowed. (It’s recommended to reload your window or restart Pulsar after changing this setting.)

---

(I've made this change locally, so it'll get addressed once I address any further feedback you have.)

---

Do you have these changes ready to go? I'll probably end up testing the bins and tap out of reading the big commit's message and diff. And everything else seems fine to my eyes. So, I probably won't have much further or detailed feedback beyond this point.